src/main/java/org/plumelib/bcelutil/StackMapUtils.java

```diff
@@ -630,6 +640,188 @@ protected final void update_uninitialized_NEW_offsets(InstructionList il) {
+    }
+    LocalVariableGen new_lvg =
+        mgen.addLocalVariable(
+            "DaIkOnTeMp" + offset, live_range_type, offset, live_range_start, live_range_end);
+    debug_instrument.log(
+        "Added local  %s, %d, %d : %s, %s%n",
+        new_lvg.getIndex(),
+        new_lvg.getStart().getPosition(),
+        new_lvg.getEnd().getPosition(),
+        new_lvg.getName(),
+        new_lvg.getType());
+    live_range_operand_size = Math.min(live_range_operand_size, live_range_type.getSize());
+  }
+
+  /**
+   * Calculate the live range of a local variable (or variables).
+   *
+   * @param mgen MethodGen of method to search
+   * @param offset offset of the local
+   * @return minimum size of local(s) found at offset
+   */
+  protected final int gen_locals_from_byte_codes(MethodGen mgen, int offset) {
+    // The same local offset could be used for multiple local variables
+    // with disjoint lifetimes.  We attempt to deal with this by looking
+    // at the type of a store instruction and if it does not equal the
+    // current type we create a temp from the current live range data
+    // and then reset to start a new live range.
+
+    // reset globals
+    live_range_start = null;
+    live_range_end = null;
+    live_range_type = null;
+    // only sizes are 1 or 2; start with something larger.
+    live_range_operand_size = 3;
+    return gen_locals_from_byte_codes(mgen, offset, mgen.getInstructionList().getStart());
+  }
+
+  /**
+   * Calculate the live range of a local variable starting from the given InstructionHandle. The
+   * following live_range globals must be set:
+   *
+   * <ul>
+   *   <li>live_range_start
+   *   <li>live_range_end
+   *   <li>live_range_type
+   *   <li>live_range_operand_size
+   * </ul>
+   *
+   * @param mgen MethodGen of method to search
+   * @param offset offset of the local
+   * @param start search forward from this instruction
+   * @return minimum size of local(s) found at offset
+   */
+  protected final int gen_locals_from_byte_codes(
+      MethodGen mgen, int offset, InstructionHandle start) {
+    OperandStack stack;
+    get_method_stack_types(mgen);
+    InstructionList il = mgen.getInstructionList();
+    for (InstructionHandle ih = start; ih != null; ih = ih.getNext()) {
+      Instruction inst = ih.getInstruction();
+
+      debug_instrument.log(
+          "gen_locals_from_byte_codes for offset: %d :: position: %d, inst: %s%n",
+          offset, ih.getPosition(), inst);
+
+      if (inst instanceof StoreInstruction) {
+        if (offset != ((LocalVariableInstruction) inst).getIndex()) {
+          continue;
+        }
+        stack = stack_types.get(ih.getPosition());
+        // get type of item about to be stored
+        Type tos = stack.peek(0);
+        // System.out.printf ("tos: %s, live_type: %s%n", tos, live_range_type);
+        // Store of a null does not change type.
+        // UNDONE: if tos is super of live_range_type, should not start new range
```

Comment:
Should "super" be "sub"?

Reply:
yes

Reply:
I made this change.  In general, if you agree with a comment, please act on it and click "resolve conversation".  There is no need to respond to every comment with a reply; doing so just clutters the conversation.  Thanks!